### DISCUSSION TITLE
Safari on iOS: Downloading large albums (more than 100 photos) fails

### DISCUSSION THREAD

**Discussion by THF-Mediapics** [photoprism/photoprism#4668] [Q&A]
I have encountered an issue where Safari on iOS fails to download large albums containing more than 100 photos. Smaller albums (e.g., 40 photos) download without any issues.

Observed Behavior:
	•	When trying to download an album with more than 100 photos in Safari on iOS, the following error message appears:
	“Safari cannot open the page. Error: The operation could not be completed. (WebKitBlobResource error 1.)”
	•	This issue occurs only in Safari on iOS. Other browsers, such as Google Chrome on the same device, download the album successfully.

Additional Confusion in Safari:
	•	In Safari, the error message appears after a short time, and there is no visible indication that the download is being prepared in the background. This lack of feedback leads to confusion, as users might think nothing is happening.
	•	In contrast, Google Chrome behaves differently: It downloads the files in the background and then, after approximately 2–3 minutes, prompts the user to choose a location to save the files. At this point, the download is essentially already complete.

Configuration:
	•	Photoprism Version: PhotoPrism® Plus Build 240915-e1280b2fb
	•	Docker: Yes (on Synology NAS)
	•	Server-Side ZIP Download: PHOTOPRISM_DOWNLOAD_ZIP=true
	•	Temporary Directory: /photoprism/tmp with sufficient storage space
	•	Reverse Proxy Timeout: 600 seconds

Expected Behavior:
	•	Downloading an album should work reliably in Safari on iOS, regardless of the number of photos.
	•	There should also be some visible feedback or progress indication in Safari during the preparation of the download to avoid user confusion.

Steps to Reproduce:
	1.	Create an album in Photoprism with more than 100 photos.
	2.	Attempt to download the album using Safari on an iOS device.
	3.	Observe that the error message “WebKitBlobResource error 1” appears after a short time.
	4.	Compare this to Google Chrome: Chrome downloads the album in the background and only prompts for a save location after 2–3 minutes, once the download is nearly complete.

Tested Workarounds:
	•	Google Chrome on iOS works flawlessly, which suggests that the issue is specific to Safari.
	•	Smaller albums (e.g., fewer than 50 photos) can still be downloaded successfully in Safari.

Questions for the Community and Developers:
	•	Is this a known issue in Photoprism or related to Safari on iOS?
	•	Are there any workarounds or solutions to improve album downloads in Safari?
	•	If this is a Safari-specific limitation, could server-side optimizations or a visible progress indicator help improve the user experience?

Thank you for your support! I’m looking forward to ideas, potential fixes, or confirmation that this may be a bug.

**Comment by omerdduran** [comment]
Hi. It seems to be a problem caused by Safari. The "WebKitBlobResource" error in Safari usually occurs when the browser engine has problems processing blob URLs. These are special URLs that refer to large amounts of data stored directly in the browser (e.g. when downloading files from web applications). You can look at this issue for improving the user experience

- https://github.com/photoprism/photoprism/issues/4341

https://communities.apple.com/de/thread/255060318?sortBy=rank
